tests/framework/pvc.go

```diff
@@ -370,7 +370,6 @@ func (f *Framework) NewPodWithPVC(podName, cmd string, pvc *k8sv1.PersistentVolu
 		Spec: k8sv1.PodSpec{
 			// this may be causing an issue
 			TerminationGracePeriodSeconds: &[]int64{10}[0],
-			RestartPolicy:                 k8sv1.RestartPolicyNever,
```

Comment:
Any reason to remove this?

Reply:
Yeah, I am trying to find out why we have it in the first place, seems to date back 5+ years
basically means verifier won't attempt to restart on transient errs

I have some tests where the verifier pod regularly hits an error without much context,
and does not recover from it

Reply:
Changed to OnFailure so it only restarts if it exits with a return code other than 0
(default is Always - pod must always be running)

Reply:
I think it makes perfect sense to change it to `RestartPolicyOnFailure`, but like you, I lack some context on why we had `RestartPolicyNever` before. If it's not breaking anything I'm for the change.

Reply:
What transient errors are we thinking. we would like to know if the verifier pod errors (like maybe the file we are looking for doesn't exist). This would simply restart the failed command, which may or may not fail again depending on if the error is transient or a real issue. If we do this, we need to make sure there are some time limits on the commands we run in the verifier pod.

Reply:
I am hitting errors in container startup (not in executing commands on it)
Seems to be related to quick recreation of the deleted verifier pod, but not sure, error does not say much.

Is there a reason to go with `RestartPolicyNever`?

Reply:
We can try with the on failure policy, we will know if something breaks due to it, there is an overall suite timeout anyway.